workflow/artifacts/s3/s3.go

```diff
@@ -103,6 +105,20 @@ func (s3Driver *S3ArtifactDriver) Save(path string, outputArtifact *wfv1.Artifac
 				log.Warnf("Failed to test if %s is a directory: %v", path, err)
 				return false, nil
 			}
+
+			bucketExists, err := s3cli.BucketExists(outputArtifact.S3.Bucket)
```

Comment:
hmm, do we want to do this automatically or opt-in? @jessesuen ?

Reply:
In general, this is escalating S3 privileges that current IAM roles that workflows may not have (i.e. a breaking change). I don't think this should be done automatically.

Reply:
I have modified this so the user needs to specify it explicitly now using the `createBucketIfNotPresent` options, for each specific output parameter. (where region and objectLocking are optional; if they are not provided, the minio client defaults to 'us-east-1' and 'false' respectively)
```
      s3:
          endpoint: minio:9000
          bucket: bucket-that-does-not-exist

          createBucketIfNotPresent:
              enabled: true
              region: 'us-east-1'
              objectLocking: false
```